Synview-frontend/src/Api_handler.ts

```diff
@@ -0,0 +1,60 @@
+const url = import.meta.env.VITE_URL;
+type RegisterData = {
+  username: string;
+  email: string;
+  password: string;
+};
```

Comment:
One of the benefits of a monorepo in which you have a frontend and backend in the same codebase is you can avoid duplication of types that can drift. I'd suggest placing your zod types in `shared` or `common` top level directory; you can get the TypeScript type useing `z.infer` on a Zod schema.